Amend filter list screen dividers

rossanafmenezes opened the pull request:
Fixes issue #3258

## Changes

- Removes the drop shadow from the appbar in the filters screens
- removes the margin for the dividers on the filter screens

## Screenshots

| Before | After |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/30724184/130511153-70d7026a-4a2b-4916-86f1-887c0a92302e.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/130511168-b7bc23ef-e847-4a80-8ced-d35b79853943.png" width="350"/> |
| <img src="https://user-images.githubusercontent.com/30724184/130511264-72689cd1-1075-420b-8e46-bb2d288725b8.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/130511276-7a517053-189a-470e-9f93-ff437f4ac591.png" width="350"/> |
| <img src="https://user-images.githubusercontent.com/30724184/130511389-6902318d-7ef3-4aab-a899-dc88a930b14b.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/130511397-56ded044-0b0c-473d-b3d9-d640a4f29db9.png" width="350"/> |
| <img src="https://user-images.githubusercontent.com/30724184/130512543-b425b384-2852-4418-879e-84af5079d978.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/130512562-cbbd0a3c-d22b-49b5-8a64-a9411d0a174f.png" width="350"/> |

## Testing Instructions

- Open the app
- Go to the Products screen
- Click on Filters and see the shadow has been removed from the appbar/ dividers are now full bleed
- Click on any of the filters and see the same result as mentioned above

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

rossanafmenezes committed:
remove margins on filter screen divider

remove margins on filter screen dividers to make them full-bleed as per Figma

kidinov committed:
Fixed wrong formatting

rossanafmenezes committed:
change appbar elevation

Changes appbar_elevation on the Filter's screen from 4dp to 2dp

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductFilterListFragment.kt:
```diff
@@ -55,8 +55,7 @@ class ProductFilterListFragment :
             addItemDecoration(
                 AlignedDividerDecoration(
```
If we don't align divider anymore, do we still need to use here `AlignedDividerDecoration` or maybe simple
 `DividerItemDecoration` should work too?

rossanafmenezes replied:
Yup, it works and makes way more sense 😃 Thanks for that.

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:
```diff
@@ -415,7 +415,7 @@ class MainActivity :
             showCrossIcon = false
         } else {
-            binding.appBarLayout.elevation = resources.getDimensionPixelSize(R.dimen.appbar_elevation).toFloat()
+            binding.appBarLayout.elevation = resources.getDimensionPixelSize(R.dimen.appbar_elevation_filter).toFloat()
```
So we want to show 2dp elevation of app bar on all not "top level" fragments, correct? I think then calling the dimen with `filter` at the end might be confusing. Maybe to call it `appbar_elevation_secondary` or something like that?

rossanafmenezes replied:
Good question.

Let me confirm with Garance.

kidinov reviewed (commented):
Looks good! Thanks! I left a couple of minor comments/questions. Please take a look

rossanafmenezes committed:
Amending files to reflect reviewer suggestions

Changed the dimens Item name from appbar_elevation_filter to appbar_elevation_secondary

Changed AlignedDividerDecoration to DividerItemDecoration since we are no longer aligning the divider anymore

rossanafmenezes commented:
> Looks good! Thanks! I left a couple of minor comments/questions. Please take a look

Thank you @kidinov 

I have made the changes to reflect what you suggested. Thank you for reviewing it and for the input 😃 

Just need confirmation from Garance regarding the Non-top-level screens question. Thank you for bringing this up

rossanafmenezes committed:
Changed formatting due to lint error

removed unused imports and reformatted one line that was too long.

Garance91540 commented:
The expected behavior is to display the top nav without dropshadow by default, and with dropshadow when the screen is scrolled. It seems like it's well implemented on the other screens so we can only focus it on this Filters screen.

kidinov commented:
@rossanafmenezes It looks like that the issue might be getting more complicated. If you want you can split the issue on 2:
* Shadow related 
* Dividers

but it's totally up to you

rossanafmenezes commented:
@kidinov sounds like a plan to me. I will revert the commit on the shadow-related code and leave this PR for the dividers and we can create a separate one for the shadow. 😃

rossanafmenezes committed:
Reverting the change for the appbar shadow

Splitting up this PR as discussed. Reverting the changes on the appbar shadow and leaving only the dividers to be merged for now.

rossanafmenezes commented:
All done @kidinov  😃

kidinov reviewed (approved):
Looks good!
